options.go

```
@@ -55,14 +62,50 @@ func WithMetrics(m Metrics) Option {
 }
 
 // WithWorkers configures the PHP workers to start
-func WithWorkers(name string, fileName string, num int, env map[string]string, watch []string) Option {
+func WithWorkers(name string, fileName string, num int, options ...WorkerOption) Option {
 	return func(o *opt) error {
-		o.workers = append(o.workers, workerOpt{name, fileName, num, PrepareEnv(env), watch})
-
+		worker := workerOpt{
+			name:                   name,
+			fileName:               fileName,
+			num:                    num,
+			env:                    PrepareEnv(nil),
+			watch:                  []string{},
+			maxConsecutiveFailures: DefaultMaxConsecutiveFailures,
+		}
+
+		for _, option := range options {
+			option(&worker)
+		}
+
+		o.workers = append(o.workers, worker)
 		return nil
 	}
 }
 
+// WithWorkerEnv sets environment variables for the worker
+func WithWorkerEnv(env map[string]string) WorkerOption {
+	return func(w *workerOpt) {
+		w.env = PrepareEnv(env)
+	}
+}
+
+// WithWorkerWatchMode sets directories to watch for file changes
+func WithWorkerWatchMode(watch []string) WorkerOption {
+	return func(w *workerOpt) {
+		w.watch = watch
+	}
+}
+
+// WithWorkerMaxFailures sets the maximum number of consecutive failures before panicking
+func WithWorkerMaxFailures(maxFailures int) WorkerOption {
+	return func(w *workerOpt) {
+		if maxFailures < -1 {
+			maxFailures = DefaultMaxConsecutiveFailures
```

We should return an error here instead of silently ignoring the invalid value